bin/console

```diff
@@ -34,6 +34,13 @@
  * ---------------------------------------------------------------------
  */
 
+// Try detecting if we are running with the root user (Not available on Windows)
+if (function_exists('posix_geteuid') && posix_geteuid() === 0) {
+    echo "\t" . __('ERROR: running as root is not allowed') . "\n";
+    echo "\t" . __('Run the script as the same user that your web server runs as') . "\n";
+    exit(1);
+}
```

Comment:
Instead of blocking the execution, unless the execution is made in "non-interactive" mode, we should add a warning and then a question like "Do you want to continue? [yes/No]".

Reply:
I agree; it's certainly not a good idea to run everything as root, but we should not block. More globally, I guess the warning should be displayed if current user is not the web server one (not only root).

Reply:
I don't know of any good way to say for certain which user is the one that the web server is running as. Plus, the root user is really the only one I see causing issues and I don't see any legitimate reason for GLPI CLI or Cron to ever need to be run as root (unless the web server itself is and that's not a behavior to be encouraged).

If I run a CLI command as my own account, I just get an access denied error on the cache or logs files; Nothing gets broken.
If I run a CLI command with `sudo` and not specifying another user, the command works but also breaks permissions on some of the files it touches.

There is a group of GLPI administrators that have very little knowledge regarding both web servers and Linux and I guess that accidentally (or purposefully without knowing the consequences) running GLPI CLI commands or Cron as root is far too easy for them.

Reply:
I guess it would be possible for some non root users to cause the same problem; but yes, root is the most evident one - so OK to display warning for this one only.
But we still do not have to block - either we throw a warning, or we provide a way to bypass. I'm pretty sure some will tell us they want to use root (their choice).